File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> heap=new PriorityQueue<>();
        for(int i:nums){
            heap.add(-i);
        }
        System.out.println(heap);
        long score=0;
        for(int i=0;i<k;i++){
            int largest=-heap.poll();
            score+=largest;
            largest=(int)Math.ceil(largest/3.0);
            heap.add(-largest);
        }
        return score;
    }
}

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        heap=[]
        for i in nums:
            heapq.heappush(heap,-i)
        score=0
        for i in range(k):
            largest=-heapq.heappop(heap)
            score+=largest
            largest=math.ceil(largest/3)
            heapq.heappush(heap,-largest)
        print(heap)
        return score